### DISCUSSION TITLE
Idea List of FAQ topics

### DISCUSSION THREAD

**Discussion by Bhupesh-V** [developersIndia/wiki#21] [Ideas]
There are a bunch of things that folks in the community ask repeatedly, here are some of them summarized.

1. ~Guide on freelance (I am assuming the guide should cover almost everything, at least for devs)~
2. How to find and distinguish good resources from bad ones?
3. Learning how to read documentation. Examples of how to do so with various popular libraries / languages would be great.
4. ~How to contribute to FOSS, Hunting Issues, Best way to communicate~
5. Git workflows: branching strategies, rebasing etc.
    - Work in Progress by Wiki Team 
7. Wiki for non-engineering students to get into tech.
8. How to get an off-campus job?
9. Product Companies v/s WITCH: What to choose?
10. I am from X-tier college, how to get a job?
11. ~How to prepare for interviews (cover placements etc)?~
12. What to learn programming if I am from a non-tech background?
13. ~Negotiating Salaries~
14. How to know if I'm being micromanaged at work?
15. Changing domains after years of work experience in another domain.
16. Career Transitions
      - QA/SDET to Dev.
      - Dev to SDET/QA.

If you have any questions about a topic, ask them below, provide suggestions, you have the platform!

**Comment by blazete** [comment]
13. Ways to approach a DSA problem.

**Reply by Bhupesh-V** [reply]
Sounds like a neat idea 👌🏽
Do you have an interest in authoring this wiki?

**Reply by triach-rold** [reply]
I could help ! I'll do a basic writeup in a md file, we can commit it into the repo if we find it helpful

**Reply by triach-rold** [reply]
I submitted a pull request for the same. Finished the writeup and md formatting.

**Comment by aravindputrevu** [comment]
@Bhupesh-V I'd love to contribute to `How to contribute to FOSS, Hunting Issues, Best way to communicate`

**Reply by Bhupesh-V** [reply]
Hey @aravindputrevu thanks for the initiative. 

However, we published something similar a few days ago:

[Discovering Small GitHub Projects for Contributing to FOSS](https://wiki.developersindia.in/faqs/finding-small-foss-projects-on-github#discovering-small-github-projects-for-contributing-to-foss)

Do you think we can improve this new wiki, or maybe cover something else in another wiki? A couple of ideas related to this specific topic

- How to communicate effectively while contributing to FOSS (talk about clearing your context, a bit about writing, etc)
- How to approach debugging issues on open-source projects (talk about looking for related issues, PRs that change the files that may have the bug etc).

> The idea is to make the wiki as niche as possible (unique stuff, that is not usually read or available on the internet)

**Reply by aravindputrevu** [reply]
Hmm, that wiki looks interesting. Ok, how about something around `continuous skill building`? 

For example: Someone trying to learn topics in AI, where and how to proceed in general? 

I can also directly write about "Skills to build for AI world" - a talk I have been doing in the recent times.

**Reply by Bhupesh-V** [reply]
1. The first idea (about continuous skill building) might have to be discussed more, As in what topics we can cover in that guide. Definitely yes, but needs some refinement.
   - We covered a very high level overview on [how to upskill yourself](https://wiki.developersindia.in/community-guides/how-to-constantly-upskill-yourself-as-a-developer), in the past, but there's certainly room for improvement.
2. +1 for this. Looks necessary for the current times. Feel free to open a proposal.

**Comment by ghost** [comment]
Cool!

**Comment by nityeshaga** [comment]
13. How Indian developers land top remote jobs

**Reply by nityeshaga** [reply]
I can also author this wiki and share what i've learnt from my [podcast](https://beyondthebrackets.org/)

**Comment by prakhar-garg-dev** [comment]
14. How to read code from bigger source code repositories ? (Tips too)
